src/widget/wtracktableview.cpp

```diff
@@ -151,21 +149,20 @@ void WTrackTableView::loadTrackModel(QAbstractItemModel* model) {
         return;
     }
 
-    // If the model has not changed
-    // there's no need to exchange the headers
-    // this will cause a small GUI freeze
+    // If the model has not changed there's no need to exchange the headers
+    // which would cause a small GUI freeze
     if (getTrackModel() == trackModel) {
         // Re-sort the table even if the track model is the same. This triggers
         // a select() if the table is dirty.
         doSortByColumn(horizontalHeader()->sortIndicatorSection(),
                 horizontalHeader()->sortIndicatorOrder());
+
+        if (restoreState) {
+            restoreCurrentViewState();
+        }
```

Comment:
My last state was, that when the model is based on track model and the entries do have a ID, the state stores the trackIDs. If the model does not have ids, it saves the index.

Reply:
yes, that's why I mention it. I adopted that commit but during tests (restoring a state after I have hidden tracks)  I ran into some debug assertion in TrackDAO which I could not relate to the selection changes. (`main` changed since #3063 so it may not have occured there earlier)

The current state of this PR is a huge improvement already and I want others to review it.
Thus I will not debug the issue I encountered NOW but I definitely have those changes stashed and _will_ give it another shot at some time.